hyperbahn/test/time-series/_time-series-cluster.js

```diff
@@ -0,0 +1,720 @@
+
+TimeSeriesCluster.buildString = function buildString(size) {
+    var tempArray = [];
+
+    for (var i = 0; i < size; i++) {
+        tempArray.push('A');
+    }
+
+    return tempArray.join('');
+};
+
+module.exports = TimeSeriesCluster;
+
+function TimeSeriesCluster(opts) {
+    /* eslint max-statements: [2, 40] */
+    if (!(this instanceof TimeSeriesCluster)) {
+        return new TimeSeriesCluster(opts);
+    }
+
+    var self = this;
+
+    nodeAssert(opts && opts.buckets, 'requires buckets');
+    nodeAssert(opts && opts.clientTimeout, 'requires clientTimeout');
+    nodeAssert(opts && opts.endpointToRequest, 'requires endpointToRequest');
+    self.buckets = opts.buckets;
+    self.clientTimeout = opts.clientTimeout;
+    self.endpointToRequest = opts.endpointToRequest;
+
+    self.clientBatchDelay = opts.clientBatchDelay || 100;
+    self.numBatchesInBucket = opts.numBatchesInBucket || 5;
+    self.clientRequestsPerBatch = opts.clientRequestsPerBatch || 15;
+    self.serverInstances = opts.serverInstances || 10;
+    self.clientInstances = opts.clientInstances || 5;
+    self.requestBody = opts.requestBody || '';
+    self.requestBodyChunks = opts.requestBodyChunks || [];
+
+    self.serverServiceName = 'time-series-server';
+    self.clientServiceName = 'time-series-client';
+
+    self.namedRemotes = [];
+    for (var sIndex = 0; sIndex < self.serverInstances; sIndex++) {
+        self.namedRemotes.push(self.serverServiceName);
+    }
+    for (var cIndex = 0; cIndex < self.clientInstances; cIndex++) {
+        self.namedRemotes.push(self.clientServiceName);
+    }
+
+    self.clusterOptions = null;
+    self.setupClusterOptions(opts);
+
+    self._cluster = HyperbahnCluster(self.clusterOptions);
+    self.logger = self._cluster.logger;
+
+    self.timeWindow = TimeWindow({
+        start: Date.now(),
+
+        buckets: self.buckets,
+        interval: self.clientBatchDelay * self.numBatchesInBucket
+    });
+
+    self.batchClient = null;
+}
+
+TimeSeriesCluster.prototype.setupClusterOptions =
+function setupClusterOptions(opts) {
+    var self = this;
+
+    self.clusterOptions = opts.cluster || {};
+    if (!self.clusterOptions.size) {
+        self.clusterOptions.size = 10;
+    }
+    if (!self.clusterOptions.remoteConfig) {
+        self.clusterOptions.remoteConfig = {};
+    }
+    if (!self.clusterOptions.remoteConfig['kValue.default']) {
+        self.clusterOptions.remoteConfig['kValue.default'] = 4;
+    }
+    if (!('rateLimiting.enabled' in self.clusterOptions.remoteConfig)) {
+        self.clusterOptions.remoteConfig['rateLimiting.enabled'] = false;
+    }
+    self.clusterOptions.namedRemotes = self.namedRemotes;
+    if (!('hyperbahn.circuits' in self.clusterOptions.remoteConfig)) {
+        self.clusterOptions.remoteConfig['hyperbahn.circuits'] = {
+            period: 100,
+            maxErrorRate: 0.5,
+            minRequests: 5,
+            probation: 5,
+            enabled: false
```

Comment:
let's not replicate the default values here. should be sufficient to enabled: false, period: 100 since those are the only overrides.